.github/workflows/ci.yml

```diff
@@ -81,18 +81,38 @@ jobs:
           - name: Astropy Dev
             os: ubuntu-latest
             python-version: "3.9"
-            toxenv: py38-astropydev
+            toxenv: py39-astropydev
+
+          - name: ASDF-Astropy Dev
```

Comment:
What I had in mind for this was not to install asdf-astropy and run the asdf tests, but to install asdf and run the asdf-astropy tests.  That's done in this workflow file:

https://github.com/asdf-format/asdf/blob/03edece11933335fc53bf32f31126f2e59a1e1b4/.github/workflows/downstream.yml

which runs once a week and also manually before we make a release.